napari/layers/vectors/vectors.py

```diff
@@ -392,7 +392,9 @@ def _get_state(self):
             {
                 'length': self.length,
                 'edge_width': self.edge_width,
-                'edge_color': self.edge_color,
+                'edge_color': self.edge_color
+                if self.data.size
+                else [self._edge.current_color],
```

Comment:
This looks like a weird way of working around the issue. I think the real issue is with empty colormaps; the error I get is:

```py
File ~/git/napari/napari/utils/colormaps/standardize_color.py:291, in _convert_array_to_correct_format(colors)
    286 if colors.shape[1] == 3:
    287     colors = np.column_stack(
    288         [colors, np.ones(len(colors), dtype=np.float32)]
    289     )
--> 291 if colors.min() < 0:
    292     raise ValueError(
    293         trans._(
    294             "Colors input had negative values.",
    295             deferred=True,
    296         )
    297     )
    299 if colors.max() > 1:

File ~/venv/napari/lib/python3.10/site-packages/numpy/core/_methods.py:44, in _amin(a, axis, out, keepdims, initial, where)
     42 def _amin(a, axis=None, out=None, keepdims=False,
     43           initial=_NoValue, where=True):
---> 44     return umr_minimum(a, axis, None, out, keepdims, initial, where)

ValueError: zero-size array to reduction operation minimum which has no identity
```

Maybe @andy-sweet has some insights on the `Color` stuff that can help here?

Reply:
But if you look at `Vector` constructor You will notice that default value for `edge_color` is provided. And their semantic meaning is `current_color` 

https://github.com/napari/napari/blob/fce50f9ae249c72ffb47b8a93507f77579cedc5b/napari/layers/vectors/vectors.py#L172

Maybe the core problem is that there are hidden 2 properties under 1 variable?
Maybe `current_color`, should be passed to the constructor independently from edge color for all edges?

Reply:
You can reproduce the fundamental error (which is related to color conversion) here as follows.

```python
transform_color(np.empty((0, 4), dtype=np.float64))
```

The 0x4 array is returned by `Vectors.edge_color` and represents the colors of the vectors. For whatever reason, this empty 0x4 array errors, whereas `[]` warns and returns white. We could change `transform_color` to return some default color instead. Or maybe return `np.empty((0, 4))` but then we'll have to change `ColorManager._from_layer_kwargs` to do something in the empty case.

I see the main problem as asymmetry: the `edge_color` init-param/setter can be a single color, many colors, or a property/feature name, whereas the `edge_color` getter always returns the array of the colors for the vectors (in this case an empty array).

I think the reason that `current_color` is not a parameter in `__init__` and is not in the return value of `_get_state` is because it roughly defines the currently selected color. As selection is not an init param and is not returned by `_get_state`, then it follows that `current_color` should not be either. @kevinyamauchi may be able to provide more context or a different opinion.

I think @Czaki's fix here is pretty much the right thing to do here. An alternative fix would be to fall back to some default value in `ColorManager._from_layer_kwargs` when `color_values` is empty. 

The complex logic of `ColorManager._from_layer_kwargs` was one of my motivations (among others) for redesigning how colors (and other related things) are specified. So this issue adds extra motivation for me to try to take that work over the line, especially as I can see some other issues related to this. So I'd urge us not to add more public API to the initializer and `_get_state` and temporarily workaround this issue instead.

Reply:
> I think the reason that current_color is not a parameter in __init__ and is not in the return value of _get_state is because it roughly defines the currently selected color. As selection is not an init param and is not returned by _get_state, then it follows that current_color should not be either. @kevinyamauchi may be able to provide more context or a different opinion.

Yes, I think this captures our reasoning when we initially made the points coloring system. I agree that @Czaki 's fix makes the most sense for the current state of the codebase.

> The complex logic of ColorManager._from_layer_kwargs was one of my motivations (among others) for redesigning how colors (and other related things) are specified. So this issue adds extra motivation for me to try to take that work over the line, especially as I can see some other issues related to this. So I'd urge us not to add more public API to the initializer and _get_state and temporarily workaround this issue instead.

+1. Looking forward to color rework, @andy-sweet !